Source/Utils/BroadcastParser.cpp

```diff
@@ -0,0 +1,73 @@
+//
+//  BroadcastParser.cpp
+//  open-ephys-GUI
+//
+//  Created by Allen Munk on 4/28/23.
+//
+
+#include "BroadcastParser.h"
+#include "../Processors/GenericProcessor/GenericProcessor.h"
+
+String BroadcastParser::build(String destPlugin, String command, std::map<String, var> payload) {
+    DynamicObject::Ptr jsonObj = new DynamicObject();
+    DynamicObject::Ptr payloadObj = new DynamicObject();
+    for(auto val: payload) {
+        payloadObj->setProperty(val.first, val.second);
+    }
+    jsonObj->setProperty("plugin", destPlugin);
+    jsonObj->setProperty("command", command);
+    jsonObj->setProperty("payload", var(payloadObj));
+    String returnString = JSON::toString(var(jsonObj), true);
+    payloadObj.reset();
+    jsonObj.reset();
+    return returnString;
+}
+
+
+bool BroadcastParser::getIntField(DynamicObject::Ptr payload, String name, int& value, int lowerBound, int upperBound) {
```

Comment:
Inevitably it seems like there's going to be many different types of fields besides `int`, and interpreting the payload has more to do with the payload itself than the broadcast parser.

A more straightforward way to structure this class for developers would be to return a class encapsulating the payload, perhaps named `BroadcastPayload`, that has member methods like `getIntField(name, value, ...)`. You could then never return the raw `payload` (i.e. like in `checkForCommand`) and instead always return this wrapped payload class. This also doubles as giving us flexibility in the future to change the underlying data structure of the broadcasted payloads, as JSON is fairly slow and not amenable to large binary payloads.